___

<a href='https://www.youtube.com/@TechWithDaveworld'><img src='../TechWithDaveworldLogo.png'/></a>
___
<center><em>Content Copyright by Tech With Daveworld</em></center>

# Numbers and more in Python!

In this lecture, we will learn about numbers in Python and how to use them.

We'll learn about the following topics:

    1.) Types of Numbers in Python
    2.) Basic Arithmetic
    3.) Differences between classic division and floor division
    4.) Object Assignment in Python

## Types of numbers

Python has various "types" of numbers (numeric literals). We'll mainly focus on integers and floating point numbers.

Integers are just whole numbers, positive or negative. For example: 2 and -2 are examples of integers.

Floating point numbers in Python are notable because they have a decimal point in them, or use an exponential (e) to define the number. For example 2.0 and -2.1 are examples of floating point numbers. 4E2 (4 times 10 to the power of 2) is also an example of a floating point number in Python.

Throughout this course we will be mainly working with integers or simple float number types.

Here is a table of the two main types we will spend most of our time working with some examples:`

<table>
<tr>
    <th>Examples</th> 
    <th>Number "Type"</th>
</tr>

<tr>
    <td>1,2,-5,1000</td>
    <td>Integers</td> 
</tr>

<tr>
    <td>1.2,-0.5,2e2,3E2</td> 
    <td>Floating-point numbers</td> 
</tr>
 </table>

Basic Arithmetic

In [15]:
# Addition
2+1

3

In [17]:
# Subtraction
5-3

2

In [19]:
# Multiplication
6*2

12

In [21]:
# Division
3/2

1.5

In [23]:
# floor Division
7//4

1

In [25]:
7/4

1.75

**Whoa! What just happened? Last time I checked, 7 divided by 4 equals 1.75 not 1!**

The reason we get this result is because we are using "*floor*" division. The // operator (two forward slashes) truncates the decimal without rounding, and returns an integer result.

In [28]:
# Modulo
3%2 # Returns the remainder of the division

1

In [30]:
6%3

0

In [34]:
5%6

5

In [36]:
7%9

7

In [38]:
# powers
2**2

4

In [40]:
# Can also do root this way
4**0.5

2.0

In [42]:
# You can use parenthesis() to specify orders
(2+10) * (10+3)

156

## Variable Assignments

Now that we've seen how to use numbers in Python as a calculator let's see how we can assign names and create variables.

We use a single equals sign to assign labels to variables. Let's see a few examples of how we can do this.

In [45]:
# create an object called "a" and assign it the number 3
a = 5

In [47]:
a # calling my variable

5

In [49]:
a+a

10

In [51]:
a

5

What happens if we reassign another value to the same variable?

In [54]:
a = 10 #Reassignment 

In [56]:
#check
a

10

In [63]:
# Write a simple program to calculate tax
my_income = 100

tax_rate = 0.1

my_tax = my_income * tax_rate

In [65]:
# Show my tax
my_tax

10.0